### PR TITLE
fixed #3877. make fadeIn and fadeOut behaviours correct

### PR DESCRIPTION
**Pull request by zilongshanren**

### CODE REVIEW

File: cocos/2d/CCActionInterval.h [hunk 1]
```diff
@@ -707,84 +707,86 @@ class CC_DLL Blink : public ActionInterval
...
 protected:
-    FadeOut() {}
-    virtual ~FadeOut() {}
+    FadeIn():_reverseAction(nullptr) {}
+    virtual ~FadeIn() {}
 
 private:
-    CC_DISALLOW_COPY_AND_ASSIGN(FadeOut);
+    CC_DISALLOW_COPY_AND_ASSIGN(FadeIn);
+    CC_SYNTHESIZE(FadeTo*, _reverseAction, ReverseAction);
```
Comment (dumganhar) [inline cocos/2d/CCActionInterval.h]: Don't use `CC_SYNTHESIZE` in engine codes. It's error-prone..

File: cocos/2d/CCActionInterval.h [hunk 2]
```diff
@@ -707,84 +707,86 @@ class CC_DLL Blink : public ActionInterval
...
-    virtual ~FadeTo() {}
-    /** initializes the action with duration and opacity */
-    bool initWithDuration(float duration, GLubyte opacity);
-
-    GLubyte _toOpacity;
-    GLubyte _fromOpacity;
-
+    FadeOut():_reverseAction(nullptr) {}
+    virtual ~FadeOut() {}
+    CC_SYNTHESIZE(FadeTo*, _reverseAction, ReverseAction);
```
Comment (dumganhar) [inline cocos/2d/CCActionInterval.h]: The same here.

File: cocos/2d/CCActionInterval.h (hunk 1 above)
Comment (zilongshanren) [inline cocos/2d/CCActionInterval.h]: So I must declare setters and getter manually?

**Comment by dumganhar** [comment]
Thanks.

**Pull request merged**